### PR TITLE
Feature/rep 413 use v2 api

### PR DESCRIPTION
**Pull request by dovekie**
@alexanderyoung PTAL

### CODE REVIEW

File: bin/critterget.py [hunk 1]
```diff
@@ -120,53 +149,72 @@ def apipost_with_response_code(uri, postdata=''):
         response = requests.post(url, headers=headers, data=pdata)
         if debug:
             print 'response is {}'.format(response.text)
-        return (response.status_code, response.json())
+        return response.status_code, response.json()
 
     except requests.exceptions.RequestException as e:
-        print u'{} MessageType="ApteligentError" Apteligent API retuned an error code: {} for the call to {}.  Maybe an ENTERPRISE feature?'.format(DATETIME_OF_RUN, e, url)
+        print u'{} MessageType="ApteligentError" Apteligent API retuned an ' \
```
Comment (alexanderyoung) [inline bin/critterget.py]: Don't use \ line continuation.  Use parens instead: https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length

File: bin/critterget.py [hunk 2]
```diff
@@ -120,53 +149,72 @@ def apipost_with_response_code(uri, postdata=''):
...
+    # read app summary information.
+    # Print out in Splunk KV format.
+    # Return a dict with appId and appName.
 
-    # list of the attributes we'll hand to Apteligent, also the list we will create as Splunk Keys
-    summattrs = "appName,appType,crashPercent,dau,latency,latestAppStoreReleaseDate,latestVersionString,linkToAppStore,iconURL,mau,rating,role,appVersions"
-    atstring = "attributes=%s"%summattrs
-    gdata = apicall("apps", atstring)
+    params = {'attributes': ','.join(SUMMARY_ATTRIBUTES)}
+    summary_data = apicall("apps", params)
```
Comment (alexanderyoung) [inline bin/critterget.py]: Why the mixed quotes?  We should always use single quotes unless there is a reason not to.

File: bin/critterget.py [hunk 3]
```diff
@@ -120,53 +149,72 @@ def apipost_with_response_code(uri, postdata=''):
...
+        printstring = u'{} MessageType="AppSummary" appId={} '.format(
+            DATETIME_OF_RUN, appId
+        )
+
+        for attribute_name in SUMMARY_ATTRIBUTES:
+            printstring += u'{}="{}"'.format(
+                attribute_name,
+                summary_data[DATA][appId][attribute_name]
+            )
+            if attribute_name == "appName":
```
Comment (alexanderyoung) [inline bin/critterget.py]: I see lots of use cases for string literals here and elsewhere.  Please make an effort to clean them up.

File: bin/critterget.py [hunk 4]
```diff
@@ -815,53 +914,84 @@ def getCredentials(sessionKey):
...
+
+    if debug:
+        print u'{} MessageType="ApteligentDebug" sessionKey is {}'.format(
+            DATETIME_OF_RUN,
+            sessionKey
+        )
 
     if len(sessionKey) == 0:
-        print u'{} MessageType="ApteligentError" Did not receive a session key from splunk. '.format(DATETIME_OF_RUN)
+        print u'{} MessageType="ApteligentError" ' \
```
Comment (alexanderyoung) [inline bin/critterget.py]: Line continuation here too.

File: bin/critterget.py [hunk 5]
```diff
@@ -62,16 +63,42 @@
...
 
 def apicall_with_response_code(uri, attribs=None):
     # perform an API call
-    if (debug): print u'access token is {}'.format(access_token)
     url = BASEURL + uri
 
-    if (attribs): url += "?"+attribs
-
-    if (debug): print u'reqstring is {}'.format(url)
+    if debug:
```
Comment (alexanderyoung) [inline bin/critterget.py]: Moving debug outside of global makes this and other functions break unless main() is called first.  Why not have a global debug var?

Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py (hunk 4 above)
Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py (hunk 3 above)
Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py [hunk 6]
```diff
@@ -12,66 +13,116 @@
...
 myapp = 'crittercism_integration'
-access_token = ''
-debug = False
+
 DUMP_DIAGS = 1
 interval = 10  # minutes between runs of this script as performed by Splunk
 MAX_RETRY = 10
 
-TODAY = datetime.datetime.now() # calculate this for a common time for all summary data
+debug = os.environ.get('CR_SPLUNK_DEBUG')
```
Comment (alexanderyoung) [inline bin/critterget.py]: Should be uppercase if global

File: bin/critterget.py [hunk 7]
```diff
@@ -120,139 +173,178 @@ def apipost_with_response_code(uri, postdata=''):
...
         if retry > MAX_RETRY:
             break
 
     CrashDict = {}
     if http_code != 200:
-        print u'{} MessageType="CrashSummary" appId={} appName="{}" ' \
-              u'Could not get crash summaries for {}. ' \
-              u'Code: {} after retry {}'.format(DATETIME_OF_RUN,
+        print (u'{} MessageType="CrashSummary" appId={} appName="{}" '
+              u'Could not get crash summaries for {}. '
```
Comment (alexanderyoung) [inline bin/critterget.py]: This doesn't line up with the paren above

File: bin/critterget.py [hunk 8]
```diff
@@ -120,139 +173,178 @@ def apipost_with_response_code(uri, postdata=''):
...
+              u'Code: {} after retry {}').format(DATETIME_OF_RUN,
                                                 appId, appName,
```
Comment (alexanderyoung) [inline bin/critterget.py]: This one doesn't line up either.  Also strange we're doing one var per line except this one.

File: bin/critterget.py [hunk 9]
```diff
@@ -120,139 +173,178 @@ def apipost_with_response_code(uri, postdata=''):
...
+            printstring = (u'{} MessageType="CrashSummary" '
+                          u'appId={} appName="{}" ').format(
+                DATETIME_OF_RUN, appId, appName
+            )
+            for attribute_name in CRASH_ATTRIBUTES:
+                printstring += u'{}="{}" '.format(
+                    attribute_name,
+                    crashdata[DATA][i][attribute_name]
+                )
+                if attribute_name == "hash":
```
Comment (alexanderyoung) [inline bin/critterget.py]: Single quotes

File: bin/critterget.py [hunk 10]
```diff
@@ -120,139 +173,178 @@ def apipost_with_response_code(uri, postdata=''):
...
                                 str(data[country][city][1]),
-                                str(data[country][city][2])  )
-            print u'{} MessageType="CrashDiagsGeo" hash={} country="{}" city="{}" lat={} lon={} crashes="{}"'.format(DATETIME_OF_RUN, hash, country, city, lat, lon, crashes)
-
-
-def diag_discrete(data, hash):
+                                str(data[country][city][2]))
+            print (u'{} MessageType="CrashDiagsGeo" hash={} country="{}" '
+                  u'city="{}" lat={} lon={} crashes="{}"').format(DATETIME_OF_RUN,
+                                                                 crash_hash,
```
Comment (alexanderyoung) [inline bin/critterget.py]: Alignment

File: bin/critterget.py [hunk 11]
```diff
@@ -594,13 +699,18 @@ def getUserflowsDetails(app_id, app_name):
     :param app_name: (string) Human-readable app name
     :return: None
     """
-    uri = 'transactions/{}/details/change/P1M?pageNum=1&pageSize=10&sortBy=name&sortOrder=ascending'.format(app_id)
+    uri = 'transactions/{}/details/change/P1M'.format(app_id)
+    params = {'pageNum': 1,
+             'pageSize': 10,
```
Comment (alexanderyoung) [inline bin/critterget.py]: Alignment

File: bin/critterget.py [hunk 12]
```diff
@@ -594,13 +699,18 @@ def getUserflowsDetails(app_id, app_name):
...
+             'sortBy': 'name',
```
Comment (alexanderyoung) [inline bin/critterget.py]: String literals

File: bin/critterget.py [hunk 13]
```diff
@@ -594,13 +699,18 @@ def getUserflowsDetails(app_id, app_name):
...
+             'sortOrder': 'ascending'}
```
Comment (alexanderyoung) [inline bin/critterget.py]: String literals

File: bin/critterget.py [hunk 14]
```diff
@@ -782,25 +897,33 @@ def getCrashCounts(appId,appName):
...
-        if (debug) : print u'Attempt to get credentials {}'.format(retry)
+        if debug:
+            print u'Attempt to get credentials {}'.format(retry)
         try:
             # list all credentials
-            entities = entity.getEntities(['admin', 'passwords'], namespace=myapp,
-                                        owner='nobody', sessionKey=sessionKey)
+            entities = entity.getEntities(
+                ['admin', 'passwords'],
+                namespace=myapp,
```
Comment (alexanderyoung) [inline bin/critterget.py]: Alignment

File: bin/critterget.py (hunk 6 above)
Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py (hunk 7 above)
Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py (hunk 8 above)
Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py (hunk 9 above)
Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py (hunk 11 above)
Comment (dovekie) [inline bin/critterget.py]: fixed

File: bin/critterget.py (hunk 14 above)
Comment (dovekie) [inline bin/critterget.py]: Not sure what needs to be fixed here.

File: bin/critterget.py (hunk 13 above)
Comment (dovekie) [inline bin/critterget.py]: each of these strings only appears once, except 'name'

**Pull request merged**